src/m365/spo/commands/homesite/homesite-set.spec.ts

```diff
@@ -86,44 +91,189 @@ describe(commands.HOMESITE_SET, () => {
+    const entraGroupStub = sinon.stub(entraGroup, 'getGroupIdByDisplayName');
+    entraGroupStub.withArgs('Marketing Team').resolves('00000000-0000-0000-0000-000000000001');
+    entraGroupStub.withArgs('Sales Team').resolves('00000000-0000-0000-0000-000000000002');
+
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
+      }
+      return 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        siteUrl: siteUrl,
+        audienceNames: 'Marketing Team, Sales Team'
       }
     } as any);
-    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, requestBody);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: { IsAudiencesPresent: true, Audiences: ['00000000-0000-0000-0000-000000000001', '00000000-0000-0000-0000-000000000002'] }
+    });
+
+    entraGroupStub.restore();
+  });
+
+  it('Clears audience names', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
+      }
+      return 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        siteUrl: siteUrl,
+        audienceNames: ''
+      }
+    } as any);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: { IsAudiencesPresent: true, Audiences: [] }
+    });
+  });
+
+  it('sets the specified site as the Home Site with multiple configuration options', async () => {
+    const postRequestStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `${spoAdminUrl}/_api/SPO.Tenant/UpdateTargetedSite`) {
+        return defaultResponse;
+      }
+      return 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        siteUrl: siteUrl,
+        vivaConnectionsDefaultStart: true,
+        draftMode: false,
+        audienceIds: '00000000-0000-0000-0000-000000000001',
+        targetedLicenseType: "everyone",
+        order: 1,
+        verbose: true
+      }
+    } as any);
+
+    assert.deepStrictEqual(postRequestStub.lastCall.args[0].data, {
+      siteUrl: siteUrl,
+      configurationParam: {
+        IsAudiencesPresent: true,
+        IsInDraftModePresent: true,
+        IsVivaConnectionsDefaultStartPresent: true,
+        IsOrderPresent: true,
+        IsTargetedLicenseTypePresent: true,
+        Order: 1,
+        TargetedLicenseType: 0,
+        isInDraftMode: false,
+        vivaConnectionsDefaultStart: true,
+        Audiences: ['00000000-0000-0000-0000-000000000001']
+      }
+    });
   });
 
   it('correctly handles error when setting the Home Site', async () => {
+    const errorResponse = {
```

Comment:
Since this object is not reused, let's assign it directly to the throw statement below instead of creating a variable for it first.